src/ripple/app/misc/Manifest.h

```diff
@@ -127,6 +115,55 @@ struct Manifest
     Blob getMasterSignature () const;
 };
 
+/** Constructs Manifest from serialized string
+
+    @param s Serialized manifest string
+
+    @return `boost::none` if string is invalid
+
+    @note This does not verify manifest signatures.
+          `Manifest::verify` should be called after constructing manifest.
+*/
+/** @{ */
+boost::optional<Manifest>
+deserializeManifest(Slice s);
```

Comment:
Question about this family of functions. How about this general pattern?

1. Give `makeSlice` multiple overloads, including one that takes a `Slice` and returns it as-is (i.e. the identity function).
2. Write one implementation of `deserializeManifest`, a template, that passes its argument to `makeSlice` unconditionally.

Reply:
Dang, that’s pretty cool.

Reply:
Looking at this again, to keep the template small, we might want a "private" `detail::deserializeManifest` that is monomorphic and expects a `Slice`. All the templated function does is pass its argument through `makeSlice` to that implementation.

Reply:
I opted to not do this at this time and leave the two tiny overloads as is. Longer term, I want to restructure how this is done.